### DISCUSSION TITLE
ruby 3.2.2 compiles without readline support on linux

### DISCUSSION THREAD

**Discussion by nulty** [rbenv/ruby-build#2357] [Build failures]
Was this always the case?

I can see readline is automatically found on homebrew but I thought it would be easy enough to do the same for Linux.

I'm using mise-en-place to install via ruby-build and I just installed 3.2.2 and my non-readline issues took me here.

**Comment by nulty** [comment]
This is mise-en-place in case you're wondering

https://mise.jdx.dev/lang/ruby.html

**Comment by mislav** [comment]
> I can see readline is automatically found on homebrew but I thought it would be easy enough to do the same for Linux.

Hi, the reason ruby-build automatically discovers readline from Homebrew is that Homebrew formulae like "readline" are not automatically linked into search path for compiled programs. However, when you install readline on your Linux system using a package manager for that system (other than Homebrew), readline should be in search path and compiled programs should be able to discover it.

If you experience problems with readline Ruby extension not working even though you have readline libraries (with development headers) installed on Linux, then please grab a ruby-build log file, upload it to this thread, and provide more information about your system (such as the OS version and which command did you use to install readline). Thanks!
